src/basic/mountpoint-util.c

```diff
@@ -316,12 +318,10 @@ int fd_is_mount_point(int fd, const char *filename, int flags) {
 
 fallback_fdinfo:
         r = fd_fdinfo_mnt_id(fd, filename, flags, &mount_id);
-        if (ERRNO_IS_NEG_NOT_SUPPORTED(r) || ERRNO_IS_NEG_PRIVILEGE(r))
-                goto fallback_fstat;
         if (r < 0)
                 return r;
 
-        if (isempty(filename))
+        if (STR_IN_SET(filename, "", "."))
```

Comment:
maybe add helper

```c
static bool at_fd_is_self(const char *p) {
        return isempty(p) || STRPATH_IN_SET(".", "./");
}
```

or so?

Reply:
Seems to have a narrow use case, so for now I'd stick to a bool in this function